### ISSUE TITLE
proposal to make standalone library and add multi-instance capability

### ISSUE THREAD

**Issue by datajake1999** [nukeykt/Nuked-SC55#12]
It would be desirable to have a standalone version of the emulator that doesn't rely on the SDL library, which would make it easier to integrate into many projects. The emulator should also allow multiple instances to run in a single process.
For example, the prototype for MCU_Init would be changed from "void MCU_Init(void)" to "void MCU_Init(mcu_t *mcu)". I would do a pull request myself, however I hesitate as this project is in an early stage of development.

**Comment by pachuco** [comment]
yes, please into .lib, .a, .dll, .so.
Link and tear! Big guts dependencies put aside.

**Comment by giulioz** [comment]
Started doing a similar thing in #18

**Comment by Cacodemon345** [comment]
I don't think this is fit to be a standalone library because of the non-commercial clause in the license it uses, effectively making it completely useless for GPL-licensed projects.

**Comment by datajake1999** [comment]
That is totally understandable.

**Comment by pachuco** [comment]
But hey, could still be library shaped code AND be usable only noncommercially as per licence.
